### PR TITLE
chore: feat(auth/signer): update the signature logic

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
1. import GO SDK Demo and support new Sign method.
2. delete the `signer_helper` file (deprecated methods).

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format,
  will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: auth/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# CHANGELOG
+
+## Version 2.0.2
+
+New Features:
+
+1. Import the GO SDK with version 2.0.2.
```
Comment (ShiChangkuo) [inline auth/CHANGELOG.md:7]: suggest to add the url of user guide: https://support.huaweicloud.com/devg-apisign/api-sign-sdk-go.html

Comment (Lance52259) [reply auth/CHANGELOG.md]: done.

File: auth/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# CHANGELOG
+
+## Version 2.0.2
+
+New Features:
+
+1. Import the GO SDK with version 2.0.2, according to the [API Request Siging Guide](https://support.huaweicloud.com/devg-apisign/api-sign-sdk-go.html).
```
Comment (ShiChangkuo) [inline auth/CHANGELOG.md:7]: Signing Guide

Comment (Lance52259) [reply auth/CHANGELOG.md]: done.

**Pull request merged**